ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-isis.j2

```diff
@@ -34,30 +34,51 @@
-| {{ vlan_interface }} | {{ vlan_interfaces[vlan_interface].isis_enable }} |  {{ vlan_interfaces[vlan_interface].isis_metric | default('-') }} | {% if vlan_interfaces[vlan_interface].isis_network_point_to_point is defined and vlan_interfaces[vlan_interface].isis_network_point_to_point is not none %} point-to-point {% elif vlan_interfaces[vlan_interface].isis_passive is defined and vlan_interfaces[vlan_interface].isis_passive is not none %} passive {% else %} - {% endif %}|
+{%              if vlan_interfaces[vlan_interface].isis_enable is arista.avd.defined %}
+| {{ vlan_interface }} | {{ vlan_interfaces[vlan_interface].isis_enable }} |  {{ vlan_interfaces[vlan_interface].isis_metric | default('-') }} | {% if vlan_interfaces[vlan_interface].isis_network_point_to_point is arista.avd.defined %} point-to-point {% elif vlan_interfaces[vlan_interface].isis_passive is arista.avd.defined %} passive {% else %} - {% endif %}|
 {%              endif %}
 {%          endfor %}
 {%      endif %}
 {# ------- #}
 {# loopback interfaces #}
 {# ------- #}
-{%      if loopback_interfaces is defined and loopback_interfaces is not none %}
+{%      if loopback_interfaces is arista.avd.defined %}
 {%          for loopback_interface in loopback_interfaces | arista.avd.natural_sort %}
-{%              if loopback_interfaces[loopback_interface].isis_enable is defined and loopback_interfaces[loopback_interface].isis_enable is not none %}
-| {{ loopback_interface }} | {{ loopback_interfaces[loopback_interface].isis_enable }} |  {{ loopback_interfaces[loopback_interface].isis_metric | default('-') }} | {% if loopback_interfaces[loopback_interface].isis_network_point_to_point is defined and loopback_interfaces[loopback_interface].isis_network_point_to_point is not none %} point-to-point {% elif loopback_interfaces[loopback_interface].isis_passive is defined and loopback_interfaces[loopback_interface].isis_passive is not none %} passive {% else %} - {% endif %}|
+{%              if loopback_interfaces[loopback_interface].isis_enable is arista.avd.defined %}
+| {{ loopback_interface }} | {{ loopback_interfaces[loopback_interface].isis_enable }} |  {{ loopback_interfaces[loopback_interface].isis_metric | default('-') }} | {% if loopback_interfaces[loopback_interface].isis_network_point_to_point is arista.avd.defined %} point-to-point {% elif loopback_interfaces[loopback_interface].isis_passive is arista.avd.defined %} passive {% else %} - {% endif %}|
+{%              endif %}
+{%          endfor %}
+{%      endif %}
+{# ISIS Node-SID #}
+{%      set node_sid = namespace() %}
+{%      for loopback_interface in loopback_interfaces | arista.avd.natural_sort %}
+{%          if loopback_interfaces[loopback_interface].node_segment.ipv4_index is arista.avd.defined or loopback_interfaces[loopback_interface].node_segment.ipv6_index is arista.avd.defined %}
+{%              set node_sid.configured = true %}
```

Comment:
Instead of just testing if the loopback_interface has ipv4_index or ipv6 index, try to record those interfaces which satisfy that criteria.

like ```{%      set node_sid_los = [] %}```

After if condition: 
 ```do node_sid_los.append(loopback_interface)```

After endfor you can use logic as below, to know if there exist any loop_back with node_segment config.
```{%      if node_sid_los | length > 0 %}```

Then loop over node_sid_los: ```{% for loopback_interface in node_sid_los | arista.avd.natural_sort %}```

Here is a reference example: https://github.com/aristanetworks/ansible-avd/pull/1312/files#diff-d96a037144c1894477bdb7186bcd773a9e3ee38407a2c028b5c2626b02ca4313

Reply:
Implemented the suggested logic.